Doc/library/collections.rst

```diff
@@ -429,7 +429,7 @@ or subtracting from an empty counter.
 
     Though :class:`list` objects support similar operations, they are optimized for
     fast fixed-length operations and incur O(n) memory movement costs for
-    ``pop(0)`` and ``insert(0, v)`` operations which change both the size and
+    ``remove(0)`` and ``insert(0, v)`` operations which change both the size and
     position of the underlying data representation.
```

Comment:
It this correct that pop() doesn't take any parameter.
But remove(0) also doesn't seem right. Remove accepts the `value` to be removed, not index to be removed. Perhaps a better replacement is `remove(v)`.

I'll wait for @rhettinger's feedback on this.

Reply:
This paragraph is about `list`. `list.pop()` takes `index` parameter.

Reply:
@sir-sigurd Ah, it looks that the PR author might not have realized the context. To avoid any further confusion, we could make it a little bit more explicit:
```
 ``list.pop(0)`` and ``list.insert(0, v)`` operations which change both the size and
```

Reply:
Hi @paulhoule. Thanks for taking the time to submit a PR.

This paragraph of the documentation is comparing the deque's memory management behavior to a list's behavior when making a change in the contents of the data collection (specifically, if acting on the first element of the collection). 

Compared to a list, the advantage of a deque is memory efficiency. The deque avoids moving memory around by adding or removing data from typically the front or end of the collection. As a doubly linked list, adding or removing an item from a deque does not move any of the other elements in memory. Instead pointers to the prior or next element in the deque are updated.

For a **list**, `pop()` is equivalent to `pop(-1)` to remove the last item in a list. `pop(0)` of a list would remove the first item in the list. Unlike a deque, removing an item from the first position on the list could cause movement of items stored in memory.

[Reference to source code](https://github.com/python/cpython/blob/762f93ff2efd6b7ef0177cad57939c0ab2002eac/Modules/_collectionsmodule.c#L34)

The original paragraph is trying to illustrate that adding a new element (`insert(0, v)`) or deleting an element (`pop(0)`) to the beginning of the list (index 0) can impact where elements are stored in memory. This differs from a deque since adding an item to the beginning of the collection or removing from the beginning (`popleft()`) does not move or shift the other elements in memory. 

`remove(0)` removes the value equal to the passed parameter in this case 0. `remove(0)` is a different concept since the documentation is illustrating a point about what happens when acting on the first element of a list or deque.

To summarize, the documentation is technically accurate as written. If you would like to update the PR to clarify the concept more, please feel free to update your PR. Alternatively, please feel free to close the PR. Thanks!